adapters/adapters-base/src/main/java/org/apache/cassandra/sidecar/adapters/base/CassandraMetricsOperations.java

```diff
@@ -250,4 +295,186 @@ private List<ClientConnectionEntry> statsToEntries(Stream<ConnectedClientStats>
+    }
+
+    private Map<String, Map<String, Integer>> getPendingCompactionTasksByTable()
+    {
+        // Get pending tasks by table name from Gauge metric
+        Object value = getCompactionMetric(CompactionStatsMetrics.PENDING_TASKS_BY_TABLE_NAME);
+        return parsePendingTasksMap(value);
+    }
+
+    private Map<String, Map<String, Integer>> parsePendingTasksMap(final Object value)
+    {
+        Map<?, ?> rawMap = safeCast(value, Map.class, "pending tasks");
+        Map<String, Map<String, Integer>> result = new HashMap<>();
+
+        for (Map.Entry<?, ?> entry : rawMap.entrySet())
+        {
+            String keyspace = safeCast(entry.getKey(), String.class, "keyspace name");
+            Map<?, ?> rawTableMap = safeCast(entry.getValue(), Map.class, "table data");
+            Map<String, Integer> tableMap = new HashMap<>();
+            
+            for (Map.Entry<?, ?> tableEntry : rawTableMap.entrySet())
+            {
+                String tableName = safeCast(tableEntry.getKey(), String.class, "table name");
+                Number taskCount = safeCast(tableEntry.getValue(), Number.class, "task count");
+                tableMap.put(tableName, taskCount.intValue());
+            }
+            
+            result.put(keyspace, tableMap);
+        }
+        
+        return result;
+    }
+
+    private Object getCompactionMetric(final CompactionStatsMetrics metric)
+    {
+        String metricObjectType = String.format(METRICS_OBJ_TYPE_COMPACTION, metric.metricName());
+        return queryMetric(metricObjectType, metric.type);
+    }
+
+    private CompactionStatsResponse.CompletedCompactionsRate getCompletedCompactionsRate()
+    {
+        // Get rates from meter metric for TotalCompactionsCompleted
+        String metricObjectType = String.format(METRICS_OBJ_TYPE_COMPACTION, "TotalCompactionsCompleted");
+        MeterMetricsJmxOperations metricsProxy = jmxClient.proxy(MeterMetricsJmxOperations.class, metricObjectType);
+
+        // Convert per-second rates to the specification:
+        // meanRate: compactions per hour
+        // fifteenMinuteRate: compactions per minute for last 15 minutes
+        double meanRateValue = metricsProxy.getMeanRate() * 3600; // Convert per second to per hour
+        double fifteenMinuteRateValue = metricsProxy.getFifteenMinuteRate() * 60; // Convert per second to per minute
+
+        String meanRate = String.format("%.2f/hour", meanRateValue);
+        String fifteenMinuteRate = String.format("%.2f/minute", fifteenMinuteRateValue);
+
+        return new CompactionStatsResponse.CompletedCompactionsRate(meanRate, fifteenMinuteRate);
+    }
+
+
+    private List<ActiveCompactionEntry> getActiveCompactions(List<Map<String, String>> compactions)
+    {
+        return compactions.stream().map(compactionInfo -> {
+            // Extract fields according to specification
+            String id = compactionInfo.getOrDefault(COMPACTION_ID, DEFAULT_STRING_VALUE);
```

Comment:
not sure why we are defaulting to empty string here? should it just be null instead?